conda/plugins/types.py

```diff
@@ -420,6 +420,13 @@ class EnvironmentSpecBase(ABC):
     Base class for all env specs.
     """
 
+    # Determines if the EnvSpec plugin should be included in the set
+    # of available plugins checked during environment_spec plugin detection.
+    # If set to False, the only way to use the plugin will be through explicitly
+    # requesting it as a cli argument or setting in .condarc. By default,
+    # autodetection is enabled.
+    detection_supported: bool = True
```

Comment:
Probably a good idea to mark it as a [ClassVar](https://docs.python.org/3/library/typing.html#typing.ClassVar)

```suggestion
    detection_supported: ClassVar[bool] = True
```